pkg/operator/ceph/cluster/controller.go

```diff
@@ -502,26 +502,41 @@ func (r *ReconcileCephCluster) removeFinalizers(client client.Client, name types
 }
 
 func (r *ReconcileCephCluster) removeFinalizer(client client.Client, name types.NamespacedName, obj client.Object, finalizer string) error {
-	err := client.Get(r.opManagerContext, name, obj)
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			logger.Debugf("%s resource not found. Ignoring since object must be deleted.", name.Name)
-			return nil
-		}
-		return errors.Wrapf(err, "failed to retrieve %q to remove finalizer", name.Name)
-	}
-
-	if finalizer == "" {
-		err = opcontroller.RemoveFinalizer(r.opManagerContext, client, obj)
+	// try multiple times to remove the finalizer
+	maxTries := 3
+	var lastErr error
+	for i := 0; i < maxTries; i++ {
+		err := client.Get(r.opManagerContext, name, obj)
 		if err != nil {
-			return errors.Wrap(err, "failed to remove finalizer")
+			if kerrors.IsNotFound(err) {
+				logger.Debugf("%s resource not found. Ignoring since object must be deleted.", name.String())
```

Comment:
logger.Debugf("%s resource not found. Ignoring since object must be deleted.", name.Name)

`name types.NamespacedName` its namespaceName type so name.String() would be a combination of both namespace and name.